### PR TITLE
Add <allow_higher_versions> to 'remote' and 'auth' sections

### PR DESCRIPTION
**Pull request by nbertoldo**
## Description

As requested in #6284 the documentation for the remote and auth configuration sections are updated with new `agents/allow_higher_versions` options.

### Docs building
- [x] Compiles without warnings.
### Code formatting and web optimization
- [x] Uses three spaces indentation.
- [x] Adds or updates meta descriptions accordingly.
- [ ] Updates the `redirects.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).
### Writing style
- [x] Uses present tense, active voice, and semi-formal registry.
- [x] Uses short, simple sentences.
- [x] Uses **bold** for user interface elements, _italics_ for key terms or emphasis, and `code` font for Bash commands, file names, REST paths, and code.

### CODE REVIEW

**Review by nmkoremblum — APPROVED** [review]
LGTM

File: source/user-manual/reference/ossec-conf/auth.rst [hunk 1]
```diff
@@ -360,6 +347,25 @@ Indicates the maximum size of the queue for fetching external keys.
...
 +--------------------+------------------------------------------------------------+
 
+agents
+^^^^^^
+
+**allow_higher_versions**
+
+.. versionadded:: 4.6.0
+  
+Accept agents with a higher version than the current manager.
```
Comment (javimed) [inline source/user-manual/reference/ossec-conf/auth.rst:357]:
```suggestion
Accept agents with a later version than the current manager.
```

File: source/user-manual/reference/ossec-conf/auth.rst [hunk 2]
```diff
@@ -360,6 +347,25 @@ Indicates the maximum size of the queue for fetching external keys.
...
+
++--------------------+------------------+
+| **Default value**  | yes              |
++--------------------+------------------+
+| **Allowed values** | yes, no          |
++--------------------+------------------+
+
+.. note::
+  This option only works when **connection** is set to ``secure``.
```
Comment (javimed) [inline source/user-manual/reference/ossec-conf/auth.rst:366]:
```suggestion

   This option only works when **connection** is set to ``secure``.
```

File: source/user-manual/reference/ossec-conf/remote.rst [hunk 3]
```diff
@@ -143,6 +144,24 @@ Sets the time to close the RIDS files for agents that don't report new events in
...
 +--------------------+------------------------------------------------------------------------------------------------------------------------------------------+
 
+agents
+^^^^^^
+
+**allow_higher_versions**
+
+.. versionadded:: 4.6.0
+  
+Accept agents with a higher version than the current manager.
```
Comment (javimed) [inline source/user-manual/reference/ossec-conf/remote.rst:154]:
```suggestion
Accept agents with a later version than the current manager.
```

File: source/user-manual/reference/ossec-conf/remote.rst [hunk 4]
```diff
@@ -143,6 +144,24 @@ Sets the time to close the RIDS files for agents that don't report new events in
...
+
++--------------------+------------------+
+| **Default value**  | yes              |
++--------------------+------------------+
+| **Allowed values** | yes, no          |
++--------------------+------------------+
+
+.. note::
+  This option only works when **connection** is set to ``secure``.
```
Comment (javimed) [inline source/user-manual/reference/ossec-conf/remote.rst:163]:
```suggestion

   This option only works when **connection** is set to ``secure``.
```

**Review by javimed — APPROVED** [review]

**Review by vikman90 — CHANGES_REQUESTED** [review]
~We're missing the JSON configuration report.~
Please disregard this comment.

**Review by vikman90 — APPROVED** [review]

**Pull request merged**